scripts/script-tag.js

```diff
@@ -147,18 +151,60 @@
                     span.html(`${localDeliveryDate}`)
                 }
 
+                // if the value of the metafield is not _NA_ then only making the date field visible
                 if (metafield.value !== '_NA_') {
                     hcpreorderShipsFrom.css('visibility', 'visible');
                 }
 
                 // will handle the click event on the pre order button
                 preorderButton.on("click", addToCart);
             }
+        } else {
+            // this part executes on all the page other than product page
+            // finding an input field with name tags and then iterating over the same
+            jQueryPreOrder("input[name='tags']").map(async function (index, element) {
+                // finding a button with type submit as the button will be on the same level as the input field so using siblings
+                const preorderButton = jQueryPreOrder(element).siblings("input[type='submit']");
+                preorderButton.val(theme.strings.addToCart)
```

Comment:
Why we are using theme object here?

Reply:
Sir, to get the text of add to cart button, and whenever coming to the collections or any other page first we make it the default text for the button.

Reply:
Is it a generic solution?

Reply:
I have remove it after updating the code to simplify the logic.